types/validator_set.go

```diff
@@ -457,8 +436,7 @@ func computeNewPriorities(updates []*Validator, vals *ValidatorSet, updatedTotal
 // must have been validated with verifyUpdates() and priorities computed with computeNewPriorities().
 func (vals *ValidatorSet) applyUpdates(updates []*Validator) {
 
-	existing := make([]*Validator, len(vals.Validators))
-	copy(existing, vals.Validators)
+	existing := vals.Validators
```

Comment:
same here. why not loop over vals.Validators and apply updates? then loop over updates which add new validators.